Cg 19 20 complete

cgaddis36 opened the pull request:
Added functionality and appropriating tests for:
-Regular user profile Show Page  
-editing user profile attributes
      ~Displaying new attributes on the profile show page 

-Deleted password validation requirement in the user.rb model

cgaddis36 committed:
Created profile show page to show attributes for regular users, changed chow controller from session{:user] to take in the current_user the helper method from ApplicationController

cgaddis36 committed:
Merge branch 'master' of github.com:TravisBorgsmiller/monster_shop_1911 into sandy

cgaddis36 committed:
Merge branch 'master' of github.com:TravisBorgsmiller/monster_shop_1911 into sandy

cgaddis36 committed:
Pulled in master, editing profiles is now working and fully tested

cgaddis36 committed:
Fixed tests and editing profile functionality once again.

cgaddis36 committed:
Fixed test for edit profile

cgaddis36 committed:
changed route for profile PATCH to be more restful by taking out the /edit on the end.  Also took out inline comments on regular_profile_spec.  Also added / to route on edit profile page.

KellyIB reviewed (approved):
Still think it looks good, Chase.

TravisBorgsmiller committed:
Update regular_profile_spec.rb

Add test for blank fields to redirect back and show flash message

TravisBorgsmiller committed:
Update edit.html.erb

Cleaned up string interpolation and ensured functionality to render the previous fields info if the form has errors

TravisBorgsmiller committed:
Update users_controller.rb

Changed logic in user controller to help render previous field information and revert back to edit page upon missing fields. Also seems to work with private params just fine.

cgaddis36 reviewed (commented):
Thanks for catching the string interpolation in the link_to, sharp eye.  
Good job using the user_params in the update as well.
error_messages fix looks good too

adumortier commented on app/models/user.rb:
```diff
@@ -1,7 +1,6 @@
 class User < ApplicationRecord
 
   validates_presence_of :name, :street_address, :city, :state, :zip_code
-  validates_presence_of :password, require: true
```
this helped for the update of the password

adumortier commented on config/routes.rb:
```diff
@@ -45,6 +45,8 @@
   post "/users", to: "users#create"
   get "/profile", to: "users#show"
+  get '/profile/edit', to: "users#edit"
+  patch '/profile', to: "users#update"
```
let's ask Mike/Meg about the most restful route here
